Source/Scene/ModelUtility.js

```diff
@@ -45,6 +47,13 @@ define([
      */
     var ModelUtility = {};
 
+    // Cache the value of supportsWebp since it can take up to 300 ms to resolve.
+    var supportsWebp;
+    FeatureDetection.supportsWebp()
+    .then(function(result) {
+        supportsWebp = result;
```

Comment:
This is a race condition, anywhere that needs the value of `supportsWebp` should just be calling `FeatureDetection.supportsWebp()` instead and using the promise.

Reply:
This just makes me nervous because, should we really delay all model loading by 300 ms to wait for this check? If I can figure out a way to run this check and cache it when the application starts that seems ideal.

But yes I agree that this current implementation needs to change.

Reply:
There's no way this would be introducing a 300ms delay, it's cached the first time it's called and never a problem afterwards.  See my comment in FeatureDetection.js.